pandas/core/reshape/reshape.py

```diff
@@ -260,10 +260,13 @@ def get_new_values(self):
     def get_new_columns(self):
         if self.value_columns is None:
             if self.lift == 0:
-                return self.removed_level
+                lev = self.removed_level._shallow_copy()
```

Comment:
why wouldn't you do
```lev = self.removed_level._shallow_copy(name=self.removed_name)``` ?

Reply:
_shallow_copy and rename and other indurect methods to set the .name all call ._set_names, which does a lot of checks. Those checks are not needed in these cases, as the name has already been validated.

Perhaps have a fastpath parameter in _set_names?